common/transport/amqp/devdoc/receiver_link_requirements.md

```diff
@@ -4,71 +4,60 @@
 
 ## Public Interface
 
-### constructor(linkAddress: string, linkOptions: any, amqp10Client: amqp10.AmqpClient)
+### constructor(linkAddress: string, linkOptions: ReceiverOptions, session: Session)
 
 **SRS_NODE_AMQP_RECEIVER_LINK_16_002: [** The `ReceiverLink` class shall inherit from `EventEmitter`. **]**
 
 **SRS_NODE_AMQP_RECEIVER_LINK_16_003: [** The `ReceiverLink` class shall implement the `AmqpLink` interface. **]**
 
+**SRS_NODE_AMQP_RECEIVER_LINK_16_001: [**The `ReceiverLink` internal state machine shall be initialized in the `detached` state. **]**
+
 ### attach(callback: (err?: Error) => void): void
 
-**SRS_NODE_AMQP_RECEIVER_LINK_16_004: [** The `attach` method shall use the stored instance of the `amqp10.AmqpClient` object to attach a new link object with the `linkAddress` and `linkOptions` provided when creating the `ReceiverLink` instance. **]**
+**SRS_NODE_AMQP_RECEIVER_LINK_16_004: [** The `attach` method shall use the stored instance of the `rhea` session object to attach a new link object with the combined `linkAddress` and `linkOptions` provided when creating the `ReceiverLink` instance. **]**
 
-**SRS_NODE_AMQP_RECEIVER_LINK_16_006: [** The `ReceiverLink` object should subscribe to the `detached` event of the newly created `amqp10` link object. **]**
+**SRS_NODE_AMQP_RECEIVER_LINK_16_008: [** If the `rhea` session fails to create the link the `callback` function shall be called with this error object. **]**
 
-**SRS_NODE_AMQP_RECEIVER_LINK_16_007: [** The `ReceiverLink` object should subscribe to the `errorReceived` event of the newly created `amqp10` link object. **]**
+**SRS_NODE_AMQP_RECEIVER_LINK_16_020: [** The `attach` method shall call the `callback` if the link was successfully attached **]**
 
-**SRS_NODE_AMQP_RECEIVER_LINK_16_008: [** If the `amqp10.AmqpClient` fails to create the link the `callback` function shall be called with this error object. **]**
+**SRS_NODE_AMQP_RECEIVER_LINK_06_001: [** The `attach` method shall immediately invoke the `callback` if already in an attached state. **]**
 
-**SRS_NODE_AMQP_RECEIVER_LINK_16_020: [** The `attach` method shall call the `callback` if the link was successfully attached **]**
+**SRS_NODE_AMQP_RECEIVER_LINK_06_002: [** If the `detach` method is invoked on the `ReceiverLink` while still attaching, the ReceiverLink shall detach.  It will indicate the error to the callback for the `detach` as well as the callback to the `attach`. **]**
+
+**SRS_NODE_AMQP_RECEIVER_LINK_06_003: [** If the `forceDetach` method is invoked on the `ReceiverLink` while still attaching, the ReceiverLink shall detach.  It will indicate the error to the callback to the `attach`. **]**
```

Comment:
> It will indicate the error to the callback to the `attach`. [](start = 157, length = 60)

this sounds very high-level/generic. it should describe what is being called with what arguments.

Reply:
Changing text to: With the error supplied to the forceDetach, the `attach` callback will also be invoked.

---
In reply to: [209125261](https://github.com/Azure/azure-iot-sdk-node/pull/344#discussion_r209125261) [](ancestors = 209125261)